packages/common/src/store/buy-crypto/sagas.ts

```diff
@@ -0,0 +1,497 @@
+      )
+    }
+
+    if (amount < config.minAmount) {
+      throw new BuyCryptoError(
+        BuyCryptoErrorCode.BAD_AMOUNT,
+        `Amount is below app-configured minimum of ${
+          config.minAmount
+        } $${mint.toUpperCase()}`
+      )
+    }
+
+    if (amount > config.maxAmount) {
+      throw new BuyCryptoError(
+        BuyCryptoErrorCode.BAD_AMOUNT,
+        `Amount is above app-configured maximum of ${config.maxAmount}`
+      )
+    }
+
+    if (!outputToken) {
+      throw new BuyCryptoError(
+        BuyCryptoErrorCode.BAD_TOKEN,
+        `Unrecognized token: ${mint}`
+      )
+    }
+
+    if (!feePayerAddress) {
+      throw new BuyCryptoError(
+        BuyCryptoErrorCode.BAD_FEE_PAYER,
+        'Fee Payer is missing'
+      )
+    }
+
+    // Set up computed vars
+    const outputTokenLamports = 10 ** outputToken.decimals
+    const feePayer = new PublicKey(feePayerAddress)
+    userbank = yield* call(createUserBankIfNeeded, audiusBackendInstance, {
+      mint,
+      feePayerOverride: feePayerAddress,
+      recordAnalytics: track
+    })
+
+    // Get required SOL purchase amount via ExactOut + minRent
+    const quoteResponse = yield* call(
+      [jupiterInstance, jupiterInstance.quoteGet],
+      {
+        inputMint: TOKEN_LISTING_MAP.SOL.address,
+        outputMint: outputToken.address,
+        amount: Math.ceil(amount * outputTokenLamports),
+        swapMode: 'ExactOut',
+        slippageBps: config.slippageBps
+      }
+    )
+    const connection = yield* call(getSolanaConnection, audiusBackendInstance)
+    const minRent = yield* call(
+      [connection, connection.getMinimumBalanceForRentExemption],
+      0
+    )
+    const existingBalance = yield* call(
+      [connection, connection.getBalance],
+      wallet.publicKey
+    )
+    // otherAmountThreshold is the max input amount for the given slippage
+    const requiredAmount =
+      Number(quoteResponse.otherAmountThreshold) + minRent - existingBalance
+
+    if (requiredAmount <= 0) {
+      console.debug('SOL already sufficient for swap. Skipping purchasing step')
+    } else {
+      // Get min stripe purchase amount using USDC as quote for $1
+      const minQuote = yield* call(
+        [jupiterInstance, jupiterInstance.quoteGet],
+        {
+          inputMint: TOKEN_LISTING_MAP.SOL.address,
+          outputMint: TOKEN_LISTING_MAP.USDC.address,
+          amount: 1 * 10 ** TOKEN_LISTING_MAP.USDC.decimals,
+          swapMode: 'ExactOut',
+          slippageBps: config.slippageBps
+        }
+      )
+      const minAmount = Number(minQuote.otherAmountThreshold)
+      if (requiredAmount < minAmount) {
+        console.warn(
+          `Stripe requires minimum purchase of $1 (${minAmount} lamports). Required lamports: ${requiredAmount}`
+        )
+      }
+
+      const lamportsToPurchase = Math.max(requiredAmount, minAmount)
+
+      // Open Stripe Modal
+      // TODO: Support coinbase similarly?
```

Comment:
Yeah I think since we pass a provider, the idea would be to fire the correct modal init action, and/or we could just pass an action to this saga that invokes the correct modal with the correct properties, such that this saga has no special knowledge of providers.